lib/cPanel/TaskQueue.pm

```diff
@@ -859,6 +859,9 @@ END { undef %valid_processors }    # case CPANEL-10871 to avoid a SEGV during gl
             }
             else {
 
+                $self->_process_overrides($task);
+                next if $self->_is_duplicate_command($task);
```

Comment:
These functions operates on items in `$self->{queue_waiting}`, but we don't update `$self->{queue_waiting}` until after this loop completes.  Is it possible that two duplicates are in the deferred queue?  If so, they may both still enter the queue.

If this is a problem, we _might_ solve it with the following additional change:
```diff
-unshift @proc, $task;
+unshift @{ $self->{queue_waiting} }, $task;
```

Reply:
Good question.  I'll check.

Reply:
Yeah, I don't think we can be sure that there isn't a duplicate.  I'll resubmit with your proposed change.

Thanks for catching that.